✨ feat: Add unique constraint and index to `code` field in `BaseLocation` model

ARYAN-NIKNEZHAD opened the pull request:
### Summary

This pull request adds a unique constraint and index to the `code` field in the `BaseLocation` model. This ensures that each location has a unique code and improves query performance on the `code` field.

### Changes

- Added `unique=True` to the `code` field in the `BaseLocation` model.
- Added an index to the `code` field to enhance performance.
- Updated the database schema to reflect these changes.

### Motivation

The unique constraint ensures data integrity by preventing duplicate codes for locations. The index on the `code` field will significantly improve the performance of queries that filter or search by code, especially for large datasets.


### Related Issues

This change addresses the need for unique location codes and performance improvements in queries involving the `code` field.

### Notes

Please review the changes and provide feedback. If there are any concerns or suggestions for further improvements, feel free to modify as needed.

ARYAN-NIKNEZHAD committed:
:sparkles: feat(mixins): Add-BaseLocation-abstract-model

ARYAN-NIKNEZHAD committed:
:bulb: docs(models): Add comprehensive docstrings to location models
- Added detailed docstrings to the following models: Province, County, District, City, RuralDistrict, and Village.
- Ensured that all models have clear, informative docstrings explaining their purpose and attributes.

These changes improve the documentation within the codebase, making it easier for developers to understand the purpose and structure of each model.

ARYAN-NIKNEZHAD committed:
:bug: fix(code-review): fix the code review changes
1. Add  and
2. fix the related_name in ReralDistrict Model

ARYAN-NIKNEZHAD committed:
:bulb: docs(models): Add comprehensive docstrings to location models Closes #6

ARYAN-NIKNEZHAD committed:
:ok_hand: fix(models.py): Refactor model docstrings for clarity and conciseness

- Updated docstrings for all models to focus on their role within the app context.
- Removed detailed attribute explanations to leverage Django's built-in field documentation.
- Ensured each model's purpose and context are clearly explained.

This enhances readability and maintains concise documentation, making the codebase cleaner and easier to understand.

ARYAN-NIKNEZHAD committed:
:sparkles: feat(models): Add unique constraint and index to  field in  model

- Added unique constraint to  field in  model to ensure each location has a unique code.
- Created an index on the  field to improve query performance.

ARYAN-NIKNEZHAD commented:
This pull request closes #9. 🚀

sepehr-akbarzadeh commented on iranian_cities/models.py:
```diff
@@ -1,160 +1,181 @@
         verbose_name = _('Province')
         verbose_name_plural = _('Provinces')
-        ordering = ('id',)
+        db_table_comment = _("Model representing a province.")
```
db_table_comment must not have gettext lazy. remove it.
also add table name to sage_iranian_cities

ARYAN-NIKNEZHAD replied:
Hi @sepehr-akbarzadeh ,

Thank you for your feedback! 😊

I’ve made the following changes based on your suggestions:

Removed gettext_lazy from db_table_comment to ensure proper handling of database comments.
Added the table name to sage_table_name for clarity and consistency.

Please review the changes and let me know if there are any further adjustments needed. Your feedback is highly 
appreciated!

sepehr-akbarzadeh reviewed (changes requested):
db_table_comment must not have gettext lazy. remove it.
also add table name to sage_iranian_cities

ARYAN-NIKNEZHAD committed:
Merge branch 'master' of github.com:ARYAN-NIKNEZHAD/django-iranian-cities

ARYAN-NIKNEZHAD committed:
:ok_hand: fix(review): Remove gettext_lazy from db_table_comment and add table name

- Removed gettext_lazy from db_table_comment to ensure proper database comments.
- Added the table name to sage_iranian_cities for clarity and consistency.

ARYAN-NIKNEZHAD committed:
:ok_hand: fix(review): Remove gettext_lazy from db_table_comment and add table name

- Removed gettext_lazy from db_table_comment to ensure proper database comments.
- Added the table name to sage_name of the table for clarity and consistency.
